library/L1_Peripheral/test/adc_test.cpp

```diff
@@ -0,0 +1,97 @@
+#include "L1_Peripheral/adc.hpp"
+#include "L4_Testing/testing_frameworks.hpp"
+
+namespace sjsu
+{
+TEST_CASE("Testing ADC Interface", "[adc-interface]")
+{
+  Mock<Adc> mock_adc;
+
+  Fake(Method(mock_adc, Initialize));
+  Fake(Method(mock_adc, Read));
+  Fake(Method(mock_adc, GetActiveBits));
+  Fake(Method(mock_adc, ReferenceVoltage));
+
+  Adc & test_subject = mock_adc.get();
+
+  SECTION("GetMaximumValue")
+  {
+    // Setup
+    uint32_t expected_active_bits;
+
+    SECTION("8-bit Resolution")
+    {
+      expected_active_bits = 8;
+    }
+    SECTION("10-bit Resolution")
+    {
+      expected_active_bits = 10;
+    }
+    SECTION("12-bit Resolution")
+    {
+      expected_active_bits = 12;
+    }
+
+    When(Method(mock_adc, GetActiveBits)).AlwaysReturn(expected_active_bits);
+
+    // Exercise
+    uint32_t actual_resolution   = test_subject.GetMaximumValue();
+    uint32_t expected_resolution = (1 << expected_active_bits) - 1;
+
+    // Verify
+    Verify(Method(mock_adc, GetActiveBits)).Once();
+    CHECK(actual_resolution == expected_resolution);
+  }
+
+  SECTION("Voltage")
+  {
+    mock_adc.ClearInvocationHistory();
```

Comment:
```suggestion
```
This is not required as each Test Case and section is its own instance. Meaning, for every section, the code above is reconstructed.